### ISSUE TITLE
archive.download does not work with {sparse: true}

### ISSUE THREAD

**Issue by joehand** [mafintosh/hyperdrive#95]
archive.download never downloads if sparse is set to true. For example, in the code below `done` will never print.

```js
var memdb = require('memdb')
var hyperdrive = require('hyperdrive')
var swarm = require('hyperdrive-archive-swarm')
var raf = require('random-access-file')

var key = process.argv[2]

var drive = hyperdrive(memdb())
var archive = drive.createArchive(key, {
  sparse: true,
  file: function (name) {
    return raf('download/' + name)
  }
})
var sw = swarm(archive)
sw.once('connection', function () {
  archive.get(0, function (err, entry) {
    console.log('got metadata', entry)
    archive.download(entry, function (err) {
      if (err) throw err
      console.log('done')
    })
  })
})
```

**Comment by pfrazee** [comment]
I solved this in my code by also calling archive.content.prioritize()

**Comment by mafintosh** [comment]
fixed in 7.4.1
